infrastructure/cli/commands/helm/rollback.ts

```diff
@@ -0,0 +1,107 @@
+import chalk from 'chalk'
+
+import {helm, releaseName} from './index.js'
+import {history} from './history.js'
+import {Environment, namespace} from '../../lib/cluster.js'
+import {makeCommand} from '../../lib/command.js'
+import {confirm} from '../../lib/interactive.js'
+import {Verbosity} from '../../lib/misc.js'
+import {EnvironmentOptions} from '../../lib/options.js'
+
+type ExactRevision = {
+  revision: number
+}
+type DeltaRevision = {
+  steps: number
+}
+type Revision = DeltaRevision | ExactRevision
+
+function coerceRevision(revision: string): Revision {
+  if (revision[0] === '-') {
+    const steps = parseInt(revision.slice(1), 10)
+    if (!isFinite(steps) || steps === 0) {
+      throw new Error('Relative revision must be a whole number of steps')
+    }
+    return {steps}
+  }
+  const revisionNum = parseInt(revision, 10)
+  if (!isFinite(revisionNum)) {
+    throw new Error('Revision must be a whole number')
+  }
+  return {
+    revision: revisionNum,
+  }
+}
+
+export const {command, description, builder} = makeCommand(
+  import.meta.url,
+  'Roll back to a previous release',
+  {
+    requiredPositionals: [
+      {
+        name: 'revision',
+        description: 'Revision number or number of steps to move (e.g. -2 to go back two revisions)',
+        type: 'string',
+        coerce: coerceRevision,
+      },
+    ],
+  }
+)
+
+interface Options extends EnvironmentOptions {
+  revision: Revision
+}
+
+export async function handler({environment, revision, verbosity}: Options): Promise<void> {
+  const releases = await history(environment)
```

Comment:
It's interesting that this constant is called "releases" but [`history()` returns instances of `History`](https://github.com/ministryofjustice/manage-my-prison/pull/95/files#diff-1c2457d56090d612f6556f12c604f2bfe428f15fb8931f87fabb552d87b2c686R45) - it makes me wonder if we can get rid of the `History` type as it's kinda a `Release` without the chart isn't it?

Is this a stupid idea tho?

Reply:
that's pretty much what i've just done – means there's no need for the silly `ReturnType<typeof history> extends Promise<(infer T)[]> ? T : never` :) about to push